src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -362,24 +362,50 @@ namespace ts {
 
         setImmediate(callback: TimeOutCallback, time: number, ...args: any[]) {
             return this.immediateCallbacks.register(callback, args);
-        };
+        }
 
         clearImmediate(timeoutId: any): void {
             this.immediateCallbacks.unregister(timeoutId);
-        };
+        }
 
+        writeFile(fileName: string, content: string) {
+            const path = this.toPath(fileName);

```

Comment:
this is not completely corrects since it won't fail if containing folder does not exist. I've added [similar functionality](https://github.com/Microsoft/TypeScript/blob/tsserver-typings/src/harness/unittests/tsserverProjectSystem.ts#L423-L445) in typings acquisition work: